django/db/models/fields/related_descriptors.py

```diff
@@ -573,6 +573,12 @@ def _apply_rel_filters(self, queryset):
             queryset._known_related_objects = {self.field: {self.instance.pk: self.instance}}
             return queryset
 
+        def _remove_prefetched_objects(self):
+            try:
+                self.instance._prefetched_objects_cache.pop(self.field.related_query_name())
+            except (AttributeError, KeyError):
```

Comment:
There are existing instances in the code where a KeyError is caught when accessing elements in the `_prefetched_objects_cache` dict (see the `get_queryset()` methods). 

Removing KeyError from those other methods did indeed break tests. So having a non-existent key appears to be a real problem.